<!doctype html>
<html lang="en">
<head>
  <meta charset="utf-8" />
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <title>Chapter 3 Mapping a single cell RNAseq dataset | 4.1-Mapping_scRNAseq</title>

    <meta name="author" content="Erik Bot &amp; Asia Zonca" />
  
   <meta name="description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />
   <meta name="generator" content="placeholder" />
  <meta property="og:title" content="Chapter 3 Mapping a single cell RNAseq dataset | 4.1-Mapping_scRNAseq" />
  <meta property="og:type" content="book" />
  
  
  <meta property="og:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content="Chapter 3 Mapping a single cell RNAseq dataset | 4.1-Mapping_scRNAseq" />
  
  <meta name="twitter:description" content="<p>This is a minimal example of using the bookdown package to write a book.
The HTML output format for this example is bookdown::bs4_book,
set in the _output.yml file.</p>" />
  
  <!-- JS -->
  <script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script>
  <script src="https://kit.fontawesome.com/6ecbd6c532.js" crossorigin="anonymous"></script>
    <script src="libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />
    <link href="libs/bootstrap-4.6.0/bootstrap.min.css" rel="stylesheet" />
    <script src="libs/bootstrap-4.6.0/bootstrap.bundle.min.js"></script>
    <script src="libs/bs3compat-0.9.0/transition.js"></script>
    <script src="libs/bs3compat-0.9.0/tabs.js"></script>
    <script src="libs/bs3compat-0.9.0/bs3compat.js"></script>
    <link href="libs/bs4_book-1.0.0/bs4_book.css" rel="stylesheet" />
    <script src="libs/bs4_book-1.0.0/bs4_book.js"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script>

  <!-- CSS -->
  <style type="text/css">
    
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  </style>
      <link rel="stylesheet" href="style.css" />
  
</head>

<body data-spy="scroll" data-target="#toc">

<div class="container-fluid">
<div class="row">
  <header class="col-sm-12 col-lg-3 sidebar sidebar-book">
    <a class="sr-only sr-only-focusable" href="#content">Skip to main content</a>

    <div class="d-flex align-items-start justify-content-between">
      <h1>
        <a href="index.html" title="">4.1-Mapping_scRNAseq</a>
      </h1>
      <button class="btn btn-outline-primary d-lg-none ml-2 mt-1" type="button" data-toggle="collapse" data-target="#main-nav" aria-expanded="true" aria-controls="main-nav"><i class="fas fa-bars"></i><span class="sr-only">Show table of contents</span></button>
    </div>

    <div id="main-nav" class="collapse-lg">
      <form role="search">
        <input id="search" class="form-control" type="search" placeholder="Search" aria-label="Search">
      </form>

      <nav aria-label="Table of contents">
        <h2>Table of contents</h2>
        <div id="book-toc"></div>

        <div class="book-extra">
          <p><a id="book-repo" href="#">View book source <i class="fab fa-github"></i></a></li></p>
        </div>
      </nav>
    </div>
  </header>

  <main class="col-sm-12 col-md-9 col-lg-7" id="content">
<div id="mapping-a-single-cell-rnaseq-dataset" class="section level1" number="3">
<h1><span class="header-section-number">Chapter 3</span> Mapping a single cell RNAseq dataset</h1>
<p>##Data preprocessing</p>
<p>This example dataset from Kanton et al. (2019) (10.1038/s41586-019-1654-9) can be downloaded from: …
For the scope of the analyses, we pre-processed the data, removing low quality cells. We also performed proportional random sampling to ease following this tutorial, to a total of 10.000 cells. This implies the reduction of total cell number while keeping cell groups proportions.
To reproduce the figures in the paper, we used the complete dataset, processed using the <code>ACTIONet</code> pipeline up to clustering (see: <a href="https://compbio.mit.edu/ACTIONet/tutorials/intro.html" class="uri">https://compbio.mit.edu/ACTIONet/tutorials/intro.html</a>. The data were processed by using the following functions with default parameters: <em>normalize.ace</em>, <em>reduce.ace</em>, <em>runACTIONet</em>, <em>clusterCells</em>).</p>
<p>Note: quality control needs to be done specifically to your dataset. We assume that cells in the following steps have passed standard quality controls. You can refer to the Seurat pipeline here: <a href="https://satijalab.org/seurat/archive/v3.0/pbmc3k_tutorial.html" class="uri">https://satijalab.org/seurat/archive/v3.0/pbmc3k_tutorial.html</a> or to the scanpy one in Python: <a href="https://scanpy.readthedocs.io/en/stable/tutorials/basics/clustering.html" class="uri">https://scanpy.readthedocs.io/en/stable/tutorials/basics/clustering.html</a>.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb7-1" tabindex="-1"></a>ext_count_matrix <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="st">&#39;/Users/asia.zonca/Google Drive/My Drive/01ms_asia_erik_neurogenesis (1)/supplementary_data/Tutorial/sc_count_matrix.rds&#39;</span>)</span></code></pre></div>
Here we report the detailed pipeline of the standard Seurat steps required to obtain clusters, which are needed for the following analyses. It is possible to customize some parameters for the pipeline, which have to be tuned to your specific dataset.
<details>
<summary>
Show code
</summary>
<div class="sourceCode" id="cb8"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb8-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-1" tabindex="-1"></a><span class="co"># The pre-processing steps are done following the standard Seurat pipeline:</span></span>
<span id="cb8-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-2" tabindex="-1"></a><span class="fu">library</span>(Seurat)</span>
<span id="cb8-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-3" tabindex="-1"></a></span>
<span id="cb8-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-4" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">CreateSeuratObject</span>(<span class="at">counts =</span> ext_count_matrix)</span>
<span id="cb8-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-5" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">NormalizeData</span>(<span class="at">object =</span> sc_seu)</span>
<span id="cb8-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-6" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">FindVariableFeatures</span>(<span class="at">object =</span> sc_seu, <span class="at">span =</span> <span class="fl">0.3</span>)</span>
<span id="cb8-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-7" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">ScaleData</span>(<span class="at">object =</span> sc_seu, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(sc_seu))</span>
<span id="cb8-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-8" tabindex="-1"></a>numPCs <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="dv">50</span>, <span class="fu">ncol</span>(ext_count_matrix) <span class="sc">-</span> <span class="dv">1</span>)</span>
<span id="cb8-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-9" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">RunPCA</span>(<span class="at">object =</span> sc_seu, <span class="at">features =</span> <span class="fu">VariableFeatures</span>(sc_seu), <span class="at">npcs =</span> numPCs, <span class="at">approx =</span> <span class="cn">FALSE</span>)</span>
<span id="cb8-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-10" tabindex="-1"></a></span>
<span id="cb8-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-11" tabindex="-1"></a>PCs_to_use <span class="ot">&lt;-</span> (sc_seu<span class="sc">$</span>pca<span class="sc">@</span>stdev)<span class="sc">^</span><span class="dv">2</span></span>
<span id="cb8-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-12" tabindex="-1"></a>PCs_to_use <span class="ot">&lt;-</span> PCs_to_use<span class="sc">/</span><span class="fu">sum</span>(PCs_to_use)</span>
<span id="cb8-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-13" tabindex="-1"></a>PCs_to_use <span class="ot">&lt;-</span> <span class="fu">cumsum</span>(PCs_to_use)</span>
<span id="cb8-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-14" tabindex="-1"></a>PCs_to_use <span class="ot">&lt;-</span> <span class="fu">min</span>(<span class="fu">which</span>(PCs_to_use <span class="sc">&gt;=</span> <span class="fl">0.75</span>))</span>
<span id="cb8-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-15" tabindex="-1"></a></span>
<span id="cb8-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-16" tabindex="-1"></a>PCs_to_use <span class="ot">&lt;-</span> <span class="fu">min</span>(numPCs, PCs_to_use)</span>
<span id="cb8-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-17" tabindex="-1"></a></span>
<span id="cb8-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-18" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">FindNeighbors</span>(<span class="at">object =</span> sc_seu, <span class="at">dims =</span> <span class="fu">seq_len</span>(PCs_to_use))</span>
<span id="cb8-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-19" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">FindClusters</span>(<span class="at">object =</span> sc_seu, <span class="at">resolution =</span> <span class="dv">1</span>, <span class="at">algorithm =</span> <span class="dv">4</span>, <span class="at">random.seed =</span> <span class="dv">1</span>) <span class="co"># if this gives an error, try setting algorithm to 1 or check ?FindClusters</span></span>
<span id="cb8-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-20" tabindex="-1"></a></span>
<span id="cb8-21"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb8-21" tabindex="-1"></a>sc_seu <span class="ot">&lt;-</span> <span class="fu">RunUMAP</span>(sc_seu, <span class="at">dims =</span> <span class="fu">seq_len</span>(PCs_to_use))</span></code></pre></div>
</details>
<p>We can then visualize the UMAP with the clusters:</p>
<div class="sourceCode" id="cb9"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb9-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb9-1" tabindex="-1"></a><span class="fu">DimPlot</span>(sc_seu, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>, <span class="at">label =</span> <span class="cn">TRUE</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<p>Since this dataset contains cells from different ages, we can also check the ages distribution and map this information on the single cells:</p>
<div class="sourceCode" id="cb10"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb10-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-1" tabindex="-1"></a>age_annotation <span class="ot">&lt;-</span> <span class="fu">unlist</span>(<span class="fu">lapply</span>(<span class="fu">strsplit</span>(<span class="fu">colnames</span>(sc_seu), <span class="st">&#39;.&#39;</span>, <span class="at">fixed =</span> <span class="cn">TRUE</span>), <span class="cf">function</span>(i) {i[<span class="dv">2</span>]}))</span>
<span id="cb10-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-2" tabindex="-1"></a><span class="fu">names</span>(age_annotation) <span class="ot">&lt;-</span> <span class="fu">Cells</span>(sc_seu)</span>
<span id="cb10-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-3" tabindex="-1"></a></span>
<span id="cb10-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-4" tabindex="-1"></a>sc_seu<span class="sc">$</span>age <span class="ot">&lt;-</span> age_annotation</span>
<span id="cb10-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-5" tabindex="-1"></a></span>
<span id="cb10-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-6" tabindex="-1"></a>ages_palette <span class="ot">&lt;-</span> <span class="fu">viridis</span>(<span class="at">n =</span> <span class="fu">length</span>(<span class="fu">unique</span>(age_annotation)))</span>
<span id="cb10-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-7" tabindex="-1"></a><span class="fu">names</span>(ages_palette) <span class="ot">&lt;-</span> <span class="fu">unique</span>(age_annotation)</span>
<span id="cb10-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-8" tabindex="-1"></a></span>
<span id="cb10-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-9" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb10-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-10" tabindex="-1"></a>  sc_seu, </span>
<span id="cb10-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-11" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;age&quot;</span>,</span>
<span id="cb10-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-12" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb10-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-13" tabindex="-1"></a>  <span class="at">cols =</span> ages_palette</span>
<span id="cb10-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb10-14" tabindex="-1"></a>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div id="data-summary" class="section level2" number="3.1">
<h2><span class="header-section-number">3.1</span> Data summary</h2>
<p>The standard mapping procedure requires the definition of summary transcriptional profiles for each cluster, that can be obtained in two ways:</p>
<ol style="list-style-type: decimal">
<li>Pseudobulk profiles (obtained by summing all counts in a cluster together and then normalizing using Counts Per Million (CPM))</li>
<li>Expression signatures (obtained by averaging the normalized counts of all cells in a cluster and then subtracting the mean expression across all cells)</li>
</ol>
<p>Depending on the type of question, the user can choose the most suitable summary profile. The key distinction lies in the type of information captured: <strong>signatures</strong> represent relative expression, describing how a given cluster differs from the others and thus highlighting <strong>differences</strong>. In contrast, <strong>pseudobulk profiles</strong> capture <strong>absolute</strong> expression levels and are therefore more suitable for biological interpretation without emphasizing relative changes. Signatures are not suited if used in a relatively homogeneous dataset, as they will greatly enhance subtle differences which may just depicts noise.
For the following example we will use pseudobulk profiles, which is the more straightforward approach.</p>
<p>Pseudobulk profiles can be computed using the following function, which will save as a <em>SingleCellExperiment</em> object the pseudobulk counts in <code>counts</code> and the log-normalized counts per millions in <code>logcounts</code>:</p>
<div class="sourceCode" id="cb11"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb11-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-1" tabindex="-1"></a>get_pseudobulk <span class="ot">&lt;-</span> <span class="cf">function</span>(exp_matrix, membership_vector) { <span class="co">#the membership vector represents the group by which we want to compute pseudobulk, in our case cluster labels</span></span>
<span id="cb11-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-2" tabindex="-1"></a>  unique_mv <span class="ot">&lt;-</span> <span class="fu">sort</span>(<span class="fu">names</span>(<span class="fu">which</span>(<span class="fu">table</span>(membership_vector)<span class="sc">&gt;</span><span class="dv">1</span>))) </span>
<span id="cb11-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-3" tabindex="-1"></a>  pseudo_counts <span class="ot">&lt;-</span> <span class="fu">t</span>(DelayedArray<span class="sc">::</span><span class="fu">rowsum</span>(<span class="fu">t</span>(exp_matrix), membership_vector))</span>
<span id="cb11-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-4" tabindex="-1"></a>  <span class="fu">rownames</span>(pseudo_counts) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(exp_matrix)</span>
<span id="cb11-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-5" tabindex="-1"></a>  norm_counts <span class="ot">&lt;-</span> <span class="fu">log2</span>((<span class="fu">t</span>(<span class="fu">t</span>(pseudo_counts)<span class="sc">/</span>Matrix<span class="sc">::</span><span class="fu">colSums</span>(pseudo_counts, <span class="at">na.rm =</span> T))<span class="sc">*</span><span class="dv">1000000</span>) <span class="sc">+</span> <span class="dv">1</span>) <span class="co">#cpm normalization</span></span>
<span id="cb11-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-6" tabindex="-1"></a>  </span>
<span id="cb11-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-7" tabindex="-1"></a>  out <span class="ot">&lt;-</span> SingleCellExperiment<span class="sc">::</span><span class="fu">SingleCellExperiment</span>(<span class="at">assays=</span><span class="fu">list</span>(<span class="at">counts=</span>pseudo_counts, </span>
<span id="cb11-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-8" tabindex="-1"></a>                                                                <span class="at">logcounts=</span>norm_counts))</span>
<span id="cb11-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-9" tabindex="-1"></a>  <span class="cf">if</span>(<span class="fu">is.null</span>(<span class="fu">colnames</span>(out))) {</span>
<span id="cb11-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-10" tabindex="-1"></a>    <span class="fu">colnames</span>(out) <span class="ot">&lt;-</span> <span class="fu">colnames</span>(pseudo_counts)</span>
<span id="cb11-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-11" tabindex="-1"></a>    }</span>
<span id="cb11-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-12" tabindex="-1"></a>  out<span class="sc">@</span>metadata<span class="sc">$</span>cell_count <span class="ot">&lt;-</span> <span class="fu">table</span>(membership_vector)</span>
<span id="cb11-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-13" tabindex="-1"></a>  <span class="fu">return</span>(out)</span>
<span id="cb11-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-14" tabindex="-1"></a></span>
<span id="cb11-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb11-15" tabindex="-1"></a>}</span></code></pre></div>
<p>We now define the pseudobulks of the clusters that we obtained following the Seurat pipeline:</p>
<div class="sourceCode" id="cb12"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb12-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-1" tabindex="-1"></a>sc_pseudo <span class="ot">&lt;-</span> <span class="fu">get_pseudobulk</span>(<span class="at">exp_matrix =</span> sc_seu<span class="sc">@</span>assays<span class="sc">$</span>RNA<span class="sc">$</span>counts, </span>
<span id="cb12-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-2" tabindex="-1"></a>                            <span class="at">membership_vector =</span> sc_seu<span class="sc">$</span>seurat_clusters)</span>
<span id="cb12-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-3" tabindex="-1"></a><span class="co">#&gt; Warning in S4Vectors:::anyMissing(runValue(x_seqnames)): &#39;S4Vectors:::anyMissing()&#39; is deprecated.</span></span>
<span id="cb12-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-4" tabindex="-1"></a><span class="co">#&gt; Use &#39;anyNA()&#39; instead.</span></span>
<span id="cb12-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-5" tabindex="-1"></a><span class="co">#&gt; See help(&quot;Deprecated&quot;)</span></span>
<span id="cb12-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-6" tabindex="-1"></a><span class="co">#&gt; Warning in S4Vectors:::anyMissing(runValue(strand(x))): &#39;S4Vectors:::anyMissing()&#39; is deprecated.</span></span>
<span id="cb12-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-7" tabindex="-1"></a><span class="co">#&gt; Use &#39;anyNA()&#39; instead.</span></span>
<span id="cb12-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-8" tabindex="-1"></a><span class="co">#&gt; See help(&quot;Deprecated&quot;)</span></span>
<span id="cb12-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb12-9" tabindex="-1"></a><span class="fu">colnames</span>(sc_pseudo) <span class="ot">&lt;-</span> <span class="fu">paste0</span>(<span class="st">&#39;c&#39;</span>, <span class="fu">colnames</span>(sc_pseudo)) <span class="co">#to have strings and not numbers</span></span></code></pre></div>
</div>
<div id="mapping-on-the-reference-network" class="section level2" number="3.2">
<h2><span class="header-section-number">3.2</span> Mapping on the reference network</h2>
<p>Now comes the actual mapping part, which can be easily performed using the <em>mapNetwork</em> function, that requires the following main inputs:</p>
<ul>
<li><code>net</code>: the reference network to use</li>
<li><code>new_profiles</code>: the transcriptional profiles to map (either pseudobulk or signatures)</li>
</ul>
<p>Other useful inputs can be checked in the function help (?mapNetwork).</p>
<p>Since in the original paper they mention the presence of glial cell types, inhibitory and excitatory neurons, it is reasonable to map on the corticogenesis network. If your experimental design already defines the presence of only glial cells or only excitatory neurons (with their respective precursors), then it is suggested to map only on the neurogenesis and gliogenesis steps.
Another approach is to first map on the on the more general corticogenesis network and then map again on the precise lineage-specific networks, also on a subset of cells.</p>
<div class="sourceCode" id="cb13"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb13-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb13-1" tabindex="-1"></a>mapped_sc_pseudo <span class="ot">&lt;-</span> <span class="fu">mapNetwork</span>(<span class="at">net =</span> corticogenesis_sce, </span>
<span id="cb13-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb13-2" tabindex="-1"></a>                               <span class="at">new_profiles =</span> <span class="fu">logcounts</span>(sc_pseudo))</span></code></pre></div>
<p>We can now visualize the network with the mapped clusters on top:</p>
<div class="sourceCode" id="cb14"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb14-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb14-1" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>new_plot</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-16-1.png" width="672" /></p>
<p>Newly mapped clusters will locate close to the most similar reference clusters, so it is possible to observe the most similar subclasses for each mapped cluster.
By visualizing the mapped network, we can already appreciate how the clusters generally map into two different sections of the network, one close to radial glia and cycling radial glia, while the other close to immature neurons. To quantify similarity, we developed a score per subclass per cluster, using the 15 most similar reference clusters’ annotations (the number of reference clusters to use can be set by changing <code>n_nearest</code> in the <em>mapNetwork</em> function).
This score sums up to 1 and represents the weighted fraction of clusters in the neighborhood belonging to the different subclasses.
The score can be visualized with a barplot reporting on the y-axis the proportions (between 0-1) of the different subclasses, and on the x-axis the mapped clusters.</p>
<div class="sourceCode" id="cb15"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb15-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb15-1" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Barplot</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-17-1.png" width="672" /></p>
<p>To get a single label for each mapped cluster, we take the maximum of these scores. This depicts the most similar subclass among the closest neighbors.</p>
<div class="sourceCode" id="cb16"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb16-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-1" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Best.Annotation</span>
<span id="cb16-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-2" tabindex="-1"></a><span class="co">#&gt;      c1      c2      c3      c4      c5      c6      c7 </span></span>
<span id="cb16-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-3" tabindex="-1"></a><span class="co">#&gt;   &quot;eIN&quot;  &quot;miEN&quot;   &quot;eIN&quot;   &quot;cRG&quot;    &quot;RG&quot;  &quot;miEN&quot;   &quot;cRG&quot; </span></span>
<span id="cb16-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-4" tabindex="-1"></a><span class="co">#&gt;      c8      c9     c10     c11     c12     c13     c14 </span></span>
<span id="cb16-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-5" tabindex="-1"></a><span class="co">#&gt;  &quot;miEN&quot;   &quot;eIN&quot;    &quot;RG&quot;   &quot;eIN&quot;   &quot;cRG&quot;   &quot;cRG&quot;    &quot;RG&quot; </span></span>
<span id="cb16-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-6" tabindex="-1"></a><span class="co">#&gt;     c15     c16     c17     c18     c19     c20     c21 </span></span>
<span id="cb16-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-7" tabindex="-1"></a><span class="co">#&gt;    &quot;RG&quot; &quot;imNIT&quot; &quot;imNIT&quot;    &quot;RG&quot;    &quot;RG&quot;    &quot;RG&quot;   &quot;cRG&quot; </span></span>
<span id="cb16-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-8" tabindex="-1"></a><span class="co">#&gt;     c22     c23 </span></span>
<span id="cb16-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb16-9" tabindex="-1"></a><span class="co">#&gt;   &quot;eIN&quot;    &quot;RG&quot;</span></span></code></pre></div>
<p>The same quantification can be obtained for stages of maturation, either by specifying <em>color_attr = ‘Stages’</em> in the <em>mapNetwork</em> function, or by running the <em>annotateMapping</em> function, which requires the following inputs:</p>
<ul>
<li><code>net</code>: the reference network to use.</li>
<li><code>new_cor</code>: the correlation between mapped points and reference clusters, can be found here <em>mapped_object$new_cor</em>.</li>
<li><code>color_attr</code>: the annotation label to use, in this case ‘Stages’.</li>
</ul>
<p>Our mapping strategy allows the mapping of any expression matrix, thus we added a quantification of the mapping confidence, described with two diverse measures:
a. <strong>local score</strong> (<em>mapped_object<span class="math inline">\(annotation\)</span>Mapping.Confidence</em>) for each mapped point, showing average of the 15 highest correlation values that mapped point has with the reference clusters
b. <strong>global score</strong> (<em>mapped_object<span class="math inline">\(annotation\)</span>Global.Confidence</em>), describing the average confidence among all mapped point.</p>
<p>The confidence values can be accessed in the following way:</p>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-1" tabindex="-1"></a><span class="co"># per cluster confidence</span></span>
<span id="cb17-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-2" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Mapping.Confidence</span>
<span id="cb17-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-3" tabindex="-1"></a><span class="co">#&gt;        c1        c2        c3        c4        c5        c6 </span></span>
<span id="cb17-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-4" tabindex="-1"></a><span class="co">#&gt; 0.7130546 0.7547882 0.6901726 0.8477842 0.7965150 0.7947670 </span></span>
<span id="cb17-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-5" tabindex="-1"></a><span class="co">#&gt;        c7        c8        c9       c10       c11       c12 </span></span>
<span id="cb17-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-6" tabindex="-1"></a><span class="co">#&gt; 0.6281157 0.7647813 0.7314415 0.7639241 0.7194806 0.6995298 </span></span>
<span id="cb17-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-7" tabindex="-1"></a><span class="co">#&gt;       c13       c14       c15       c16       c17       c18 </span></span>
<span id="cb17-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-8" tabindex="-1"></a><span class="co">#&gt; 0.6288900 0.7215679 0.8474292 0.7706888 0.7342306 0.8037366 </span></span>
<span id="cb17-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-9" tabindex="-1"></a><span class="co">#&gt;       c19       c20       c21       c22       c23 </span></span>
<span id="cb17-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-10" tabindex="-1"></a><span class="co">#&gt; 0.6317051 0.7253773 0.5659153 0.6400742 0.5006500</span></span>
<span id="cb17-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-11" tabindex="-1"></a></span>
<span id="cb17-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-12" tabindex="-1"></a><span class="co"># global confidence</span></span>
<span id="cb17-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-13" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Global.Confidence</span>
<span id="cb17-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb17-14" tabindex="-1"></a><span class="co">#&gt; [1] 0.7162878</span></span></code></pre></div>
<p>The confidence score not only tells how similar are the mapped points to neighboring clusters, but also can help identifying profiles of low quality or cell types not represented in the resource.</p>
<p>For example, cluster 23 has a slightly lower confidence score (~0.50) compared to the others, so it may be worth further checking the quality of the cells in that cluster (<strong>not done here</strong>), to assess whether this could be a low quality group of cells.</p>
<p>The confidence score can then be applied to the inferred annotations that we defined above, as some clusters may have a high confidence, but can have neighbors with different annotations, hence inferring the right annotation can be trickier:</p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-1" tabindex="-1"></a><span class="co"># checking the annotation confidence of the first five clusters </span></span>
<span id="cb18-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-2" tabindex="-1"></a></span>
<span id="cb18-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-3" tabindex="-1"></a>mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Annotations.Confidence[<span class="fu">seq</span>(<span class="dv">1</span>,<span class="dv">5</span>)]</span>
<span id="cb18-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-4" tabindex="-1"></a><span class="co">#&gt; $c1</span></span>
<span id="cb18-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-5" tabindex="-1"></a><span class="co">#&gt;        eIN       miEN </span></span>
<span id="cb18-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-6" tabindex="-1"></a><span class="co">#&gt; 0.61520448 0.09785007 </span></span>
<span id="cb18-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-7" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-8" tabindex="-1"></a><span class="co">#&gt; $c2</span></span>
<span id="cb18-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-9" tabindex="-1"></a><span class="co">#&gt;      miEN     imNIT </span></span>
<span id="cb18-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-10" tabindex="-1"></a><span class="co">#&gt; 0.6538900 0.1008981 </span></span>
<span id="cb18-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-11" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-12" tabindex="-1"></a><span class="co">#&gt; $c3</span></span>
<span id="cb18-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-13" tabindex="-1"></a><span class="co">#&gt;        eIN      imNIT </span></span>
<span id="cb18-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-14" tabindex="-1"></a><span class="co">#&gt; 0.64261159 0.04756101 </span></span>
<span id="cb18-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-15" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-16" tabindex="-1"></a><span class="co">#&gt; $c4</span></span>
<span id="cb18-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-17" tabindex="-1"></a><span class="co">#&gt;       cRG        RG </span></span>
<span id="cb18-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-18" tabindex="-1"></a><span class="co">#&gt; 0.5631011 0.2846830 </span></span>
<span id="cb18-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-19" tabindex="-1"></a><span class="co">#&gt; </span></span>
<span id="cb18-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-20" tabindex="-1"></a><span class="co">#&gt; $c5</span></span>
<span id="cb18-21"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-21" tabindex="-1"></a><span class="co">#&gt;         RG        cRG </span></span>
<span id="cb18-22"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb18-22" tabindex="-1"></a><span class="co">#&gt; 0.74243175 0.05408324</span></span></code></pre></div>
<p>###Knwoledge transfer</p>
<p>The annotations newly defined can now be transferred to the original dataset and visualized at the single cell level.</p>
<div class="sourceCode" id="cb19"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb19-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-1" tabindex="-1"></a>new_annotation <span class="ot">&lt;-</span> mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Best.Annotation[<span class="fu">paste0</span>(<span class="st">&#39;c&#39;</span>, sc_seu<span class="sc">$</span>seurat_clusters)]</span>
<span id="cb19-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-2" tabindex="-1"></a><span class="fu">names</span>(new_annotation) <span class="ot">&lt;-</span> <span class="fu">Cells</span>(sc_seu)</span>
<span id="cb19-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-3" tabindex="-1"></a></span>
<span id="cb19-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-4" tabindex="-1"></a>sc_seu<span class="sc">$</span>annotation <span class="ot">&lt;-</span> new_annotation</span>
<span id="cb19-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-5" tabindex="-1"></a></span>
<span id="cb19-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-6" tabindex="-1"></a>corticogenesis_palette <span class="ot">&lt;-</span> <span class="fu">unique</span>(corticogenesis_sce<span class="sc">$</span>SubClass_color)</span>
<span id="cb19-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-7" tabindex="-1"></a><span class="fu">names</span>(corticogenesis_palette) <span class="ot">&lt;-</span> <span class="fu">unique</span>(corticogenesis_sce<span class="sc">$</span>SubClass)</span>
<span id="cb19-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-8" tabindex="-1"></a></span>
<span id="cb19-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-9" tabindex="-1"></a>ggpubr<span class="sc">::</span><span class="fu">ggarrange</span>(<span class="at">plotlist =</span> <span class="fu">list</span>(</span>
<span id="cb19-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-10" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb19-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-11" tabindex="-1"></a>  sc_seu, </span>
<span id="cb19-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-12" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb19-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-13" tabindex="-1"></a>  <span class="at">label =</span> <span class="cn">TRUE</span>, </span>
<span id="cb19-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-14" tabindex="-1"></a>) <span class="sc">+</span> <span class="fu">ggtitle</span>(<span class="st">&quot;clusters&quot;</span>),</span>
<span id="cb19-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-15" tabindex="-1"></a><span class="fu">DimPlot</span>(</span>
<span id="cb19-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-16" tabindex="-1"></a>  sc_seu, </span>
<span id="cb19-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-17" tabindex="-1"></a>  <span class="at">group.by =</span> <span class="st">&quot;annotation&quot;</span>,</span>
<span id="cb19-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-18" tabindex="-1"></a>  <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>,</span>
<span id="cb19-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-19" tabindex="-1"></a>  <span class="at">cols =</span> corticogenesis_palette,</span>
<span id="cb19-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb19-20" tabindex="-1"></a>)))</span></code></pre></div>
<div class="figure" style="text-align: center"><span style="display:block;" id="fig:unnamed-chunk-21"></span>
<img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-21-1.png" alt="UMAP with clusters and annotations." width="49%" />
<p class="caption">
Figure 3.1: UMAP with clusters and annotations.
</p>
</div>
<p>By looking at the UMAP labelled by cluster together with the newly annotated UMAP, we can see that the cells annotated as cRG that are very close to all the cells annotated as RG are indeed those of the cluster 4, the cluster with similar annotation values for both cRG and RG.</p>
<p>To further validate the annotation, we can also visualize the preferentially expressed genes in the relevant subclasses:</p>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-1" tabindex="-1"></a>sc_seu<span class="sc">$</span>eIN.Score <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">colMeans</span>(</span>
<span id="cb20-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-2" tabindex="-1"></a>  <span class="fu">GetAssayData</span>(sc_seu, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)[corticogenesis_pe_genes<span class="sc">$</span>eIN[<span class="fu">which</span>(corticogenesis_pe_genes<span class="sc">$</span>eIN <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))],])</span>
<span id="cb20-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-3" tabindex="-1"></a><span class="co">#&gt; Warning: The `slot` argument of `GetAssayData()` is deprecated as of</span></span>
<span id="cb20-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-4" tabindex="-1"></a><span class="co">#&gt; SeuratObject 5.0.0.</span></span>
<span id="cb20-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-5" tabindex="-1"></a><span class="co">#&gt; ℹ Please use the `layer` argument instead.</span></span>
<span id="cb20-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-6" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb20-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-7" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where</span></span>
<span id="cb20-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-8" tabindex="-1"></a><span class="co">#&gt; this warning was generated.</span></span>
<span id="cb20-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-9" tabindex="-1"></a></span>
<span id="cb20-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-10" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc_seu, <span class="at">features =</span> <span class="st">&quot;eIN.Score&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span>
<span id="cb20-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-11" tabindex="-1"></a><span class="co">#&gt; Warning: The `slot` argument of `FetchData()` is deprecated as of</span></span>
<span id="cb20-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-12" tabindex="-1"></a><span class="co">#&gt; SeuratObject 5.0.0.</span></span>
<span id="cb20-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-13" tabindex="-1"></a><span class="co">#&gt; ℹ Please use the `layer` argument instead.</span></span>
<span id="cb20-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-14" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the Seurat</span></span>
<span id="cb20-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-15" tabindex="-1"></a><span class="co">#&gt;   package.</span></span>
<span id="cb20-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-16" tabindex="-1"></a><span class="co">#&gt;   Please report the issue at</span></span>
<span id="cb20-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-17" tabindex="-1"></a><span class="co">#&gt;   &lt;https://github.com/satijalab/seurat/issues&gt;.</span></span>
<span id="cb20-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-18" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb20-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-19" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where</span></span>
<span id="cb20-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb20-20" tabindex="-1"></a><span class="co">#&gt; this warning was generated.</span></span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-22-1.png" width="672" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb21-1" tabindex="-1"></a></span>
<span id="cb21-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb21-2" tabindex="-1"></a>sc_seu<span class="sc">$</span>cRG.Score <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">colMeans</span>(</span>
<span id="cb21-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb21-3" tabindex="-1"></a>  <span class="fu">GetAssayData</span>(sc_seu, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)[corticogenesis_pe_genes<span class="sc">$</span>cRG[<span class="fu">which</span>(corticogenesis_pe_genes<span class="sc">$</span>cRG <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))],])</span>
<span id="cb21-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb21-4" tabindex="-1"></a></span>
<span id="cb21-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb21-5" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc_seu, <span class="at">features =</span> <span class="st">&quot;cRG.Score&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-22-2.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb22-1" tabindex="-1"></a></span>
<span id="cb22-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb22-2" tabindex="-1"></a>sc_seu<span class="sc">$</span>RG.Score <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">colMeans</span>(</span>
<span id="cb22-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb22-3" tabindex="-1"></a>  <span class="fu">GetAssayData</span>(sc_seu, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)[corticogenesis_pe_genes<span class="sc">$</span>RG[<span class="fu">which</span>(corticogenesis_pe_genes<span class="sc">$</span>RG <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))],])</span>
<span id="cb22-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb22-4" tabindex="-1"></a></span>
<span id="cb22-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb22-5" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc_seu, <span class="at">features =</span> <span class="st">&quot;RG.Score&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-22-3.png" width="672" /></p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb23-1" tabindex="-1"></a></span>
<span id="cb23-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb23-2" tabindex="-1"></a>sc_seu<span class="sc">$</span>imNIT.Score <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">colMeans</span>(</span>
<span id="cb23-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb23-3" tabindex="-1"></a>  <span class="fu">GetAssayData</span>(sc_seu, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)[corticogenesis_pe_genes<span class="sc">$</span>imNIT[<span class="fu">which</span>(corticogenesis_pe_genes<span class="sc">$</span>imNIT <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))],])</span>
<span id="cb23-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb23-4" tabindex="-1"></a></span>
<span id="cb23-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb23-5" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc_seu, <span class="at">features =</span> <span class="st">&quot;imNIT.Score&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-22-4.png" width="672" /></p>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb24-1" tabindex="-1"></a></span>
<span id="cb24-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb24-2" tabindex="-1"></a>sc_seu<span class="sc">$</span>miEN.Score <span class="ot">&lt;-</span> Matrix<span class="sc">::</span><span class="fu">colMeans</span>(</span>
<span id="cb24-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb24-3" tabindex="-1"></a>  <span class="fu">GetAssayData</span>(sc_seu, <span class="at">slot =</span> <span class="st">&quot;data&quot;</span>)[corticogenesis_pe_genes<span class="sc">$</span>miEN[<span class="fu">which</span>(corticogenesis_pe_genes<span class="sc">$</span>miEN <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))],])</span>
<span id="cb24-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb24-4" tabindex="-1"></a></span>
<span id="cb24-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb24-5" tabindex="-1"></a><span class="fu">FeaturePlot</span>(sc_seu, <span class="at">features =</span> <span class="st">&quot;miEN.Score&quot;</span>, <span class="at">reduction =</span> <span class="st">&quot;umap&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-22-5.png" width="672" /></p>
<p>We can see by the expression of preferentially expressed genes that indeed the annotations match the expectations, and that cluster 4 has high expression of cRG markers, confirming the annotations given by the mapping.</p>
<p>A further analysis that can be made it seeing how our manually curated Gene Ontology Biological Processes are expressed in the mapped clusters.</p>
<p>First, we have to obtain the average expression of those gene sets:</p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-1" tabindex="-1"></a>sc_seu_cortico_GOBP <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(corticogenesis_GO, <span class="cf">function</span>(g) {</span>
<span id="cb25-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-2" tabindex="-1"></a>  </span>
<span id="cb25-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-3" tabindex="-1"></a>  g <span class="ot">&lt;-</span> g[<span class="fu">which</span>(g <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))]</span>
<span id="cb25-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-4" tabindex="-1"></a>  </span>
<span id="cb25-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-5" tabindex="-1"></a>  Matrix<span class="sc">::</span><span class="fu">colMeans</span>(<span class="fu">logcounts</span>(sc_pseudo)[g,])</span>
<span id="cb25-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-6" tabindex="-1"></a>  </span>
<span id="cb25-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-7" tabindex="-1"></a>}))</span>
<span id="cb25-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-8" tabindex="-1"></a></span>
<span id="cb25-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-9" tabindex="-1"></a><span class="fu">rownames</span>(sc_seu_cortico_GOBP) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">(GO:[0-9]+</span><span class="sc">\\</span><span class="st">)$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(sc_seu_cortico_GOBP))</span>
<span id="cb25-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb25-10" tabindex="-1"></a>sc_seu_cortico_GOBP <span class="ot">&lt;-</span> sc_seu_cortico_GOBP[,<span class="fu">mixedorder</span>(<span class="fu">colnames</span>(sc_seu_cortico_GOBP))]</span></code></pre></div>
<p>Then, we can visualize the relative expression in a Heatmap</p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-1" tabindex="-1"></a>cortico_subclass_groups <span class="ot">&lt;-</span> <span class="fu">colnames</span>(corticogenesis_preferential_GO<span class="sc">$</span>GO_Biological_Process_2025<span class="sc">$</span>preferential)</span>
<span id="cb26-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-2" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;L23IT&#39;</span>, <span class="st">&#39;L35IT&#39;</span>, <span class="st">&#39;L6IT&#39;</span>, <span class="st">&#39;L6NIT&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;matEN&#39;</span></span>
<span id="cb26-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-3" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;LAMP5&#39;</span>, <span class="st">&#39;SST&#39;</span>, <span class="st">&#39;PVALB&#39;</span>, <span class="st">&#39;VIP&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;matIN&#39;</span></span>
<span id="cb26-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-4" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;imIT&#39;</span>, <span class="st">&#39;imNIT&#39;</span>, <span class="st">&#39;eEN&#39;</span>, <span class="st">&#39;miEN&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;immEN&#39;</span></span>
<span id="cb26-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-5" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;eIN&#39;</span>, <span class="st">&#39;imIN&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;immIN&#39;</span></span>
<span id="cb26-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-6" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;Opc&#39;</span>, <span class="st">&#39;Oligo&#39;</span>, <span class="st">&#39;Astro&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;maGlia&#39;</span></span>
<span id="cb26-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-7" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;Micro&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;miGlia&#39;</span></span>
<span id="cb26-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-8" tabindex="-1"></a>cortico_subclass_groups[<span class="fu">which</span>(cortico_subclass_groups <span class="sc">%in%</span> <span class="fu">c</span>(<span class="st">&#39;RG&#39;</span>, <span class="st">&#39;cRG&#39;</span>, <span class="st">&#39;IPC&#39;</span>))] <span class="ot">&lt;-</span> <span class="st">&#39;Prog&#39;</span></span>
<span id="cb26-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-9" tabindex="-1"></a></span>
<span id="cb26-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-10" tabindex="-1"></a>cortico_subclass_groups <span class="ot">&lt;-</span> <span class="fu">factor</span>(cortico_subclass_groups, <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;Prog&#39;</span>, <span class="st">&#39;immEN&#39;</span>, <span class="st">&#39;immIN&#39;</span>, <span class="st">&#39;matEN&#39;</span>, <span class="st">&#39;matIN&#39;</span>, <span class="st">&#39;maGlia&#39;</span>, <span class="st">&#39;miGlia&#39;</span>))</span>
<span id="cb26-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-11" tabindex="-1"></a></span>
<span id="cb26-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb26-12" tabindex="-1"></a>cortico_gobp_groups <span class="ot">&lt;-</span> <span class="fu">factor</span>(<span class="fu">c</span>(<span class="fu">rep</span>(<span class="st">&#39;Astro&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;Oligo&#39;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&#39;Opc&#39;</span>, <span class="dv">3</span>), <span class="fu">rep</span>(<span class="st">&#39;OligoGlia&#39;</span>, <span class="dv">2</span>), <span class="fu">rep</span>(<span class="st">&#39;matEN&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;matIN&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;immEN&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;Micro&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;RG&#39;</span>, <span class="dv">4</span>), <span class="fu">rep</span>(<span class="st">&#39;Stress&#39;</span>, <span class="dv">4</span>)), <span class="at">levels =</span> <span class="fu">c</span>(<span class="st">&#39;RG&#39;</span>, <span class="st">&#39;immEN&#39;</span>, <span class="st">&#39;matEN&#39;</span>, <span class="st">&#39;matIN&#39;</span>, <span class="st">&#39;Opc&#39;</span>, <span class="st">&#39;Oligo&#39;</span>, <span class="st">&#39;OligoGlia&#39;</span>, <span class="st">&#39;Astro&#39;</span>, <span class="st">&#39;Micro&#39;</span>, <span class="st">&#39;Stress&#39;</span>))</span></code></pre></div>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-1" tabindex="-1"></a>cortico_gobp_pe <span class="ot">&lt;-</span> corticogenesis_preferential_GO<span class="sc">$</span>GO_Biological_Process_2025<span class="sc">$</span>preferential[<span class="fu">names</span>(corticogenesis_GO),]</span>
<span id="cb27-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-2" tabindex="-1"></a><span class="fu">rownames</span>(cortico_gobp_pe) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">(GO:[0-9]+</span><span class="sc">\\</span><span class="st">)$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(cortico_gobp_pe))</span>
<span id="cb27-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-3" tabindex="-1"></a></span>
<span id="cb27-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-4" tabindex="-1"></a>ht1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb27-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-5" tabindex="-1"></a>  cortico_gobp_pe,</span>
<span id="cb27-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-6" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&#39;preferential</span><span class="sc">\n</span><span class="st">expression&#39;</span>,</span>
<span id="cb27-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-7" tabindex="-1"></a>  <span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">ncol</span>(cortico_gobp_pe) <span class="sc">*</span> <span class="fl">4.2</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb27-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-8" tabindex="-1"></a>  <span class="at">height =</span> <span class="fu">unit</span>(<span class="fu">nrow</span>(cortico_gobp_pe) <span class="sc">*</span> <span class="fl">4.2</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb27-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-9" tabindex="-1"></a>  <span class="at">show_row_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-10" tabindex="-1"></a>  <span class="at">show_column_dend =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-11" tabindex="-1"></a>  <span class="at">row_title_rot =</span> <span class="dv">0</span>,</span>
<span id="cb27-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-12" tabindex="-1"></a>  <span class="at">column_title_rot =</span> <span class="dv">90</span>,</span>
<span id="cb27-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-13" tabindex="-1"></a>  <span class="at">column_split =</span> cortico_subclass_groups,</span>
<span id="cb27-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-14" tabindex="-1"></a>  <span class="at">cluster_column_slices =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-15" tabindex="-1"></a>  <span class="at">row_split =</span> cortico_gobp_groups,</span>
<span id="cb27-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-16" tabindex="-1"></a>  <span class="at">cluster_row_slices =</span> <span class="cn">FALSE</span>,</span>
<span id="cb27-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-17" tabindex="-1"></a>  <span class="at">column_title_gp =</span> <span class="fu">gpar</span>(<span class="at">fontsize =</span> <span class="dv">0</span>)</span>
<span id="cb27-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-18" tabindex="-1"></a>)</span>
<span id="cb27-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-19" tabindex="-1"></a></span>
<span id="cb27-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-20" tabindex="-1"></a>ht2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb27-21"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-21" tabindex="-1"></a>  <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sc_seu_cortico_GOBP))),</span>
<span id="cb27-22"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-22" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&#39;scaled</span><span class="sc">\n</span><span class="st">expression&#39;</span>,</span>
<span id="cb27-23"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-23" tabindex="-1"></a>  <span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">ncol</span>(sc_seu_cortico_GOBP) <span class="sc">*</span> <span class="fl">4.2</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb27-24"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-24" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">TRUE</span>, <span class="at">right_annotation =</span> <span class="fu">rowAnnotation</span>(<span class="at">max =</span> <span class="fu">anno_barplot</span>(<span class="fu">apply</span>(sc_seu_cortico_GOBP, <span class="dv">1</span>, max), <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">&#39;black&#39;</span>)), <span class="at">annotation_name_side =</span> <span class="st">&#39;top&#39;</span>, <span class="at">annotation_name_rot =</span> <span class="dv">0</span>), <span class="at">bottom_annotation =</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">annotation =</span> mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Best.Annotation[<span class="fu">colnames</span>(sc_seu_cortico_GOBP)], <span class="at">col =</span> <span class="fu">list</span>(<span class="at">annotation =</span> corticogenesis_palette), <span class="at">show_legend =</span> <span class="cn">FALSE</span>), <span class="at">show_column_dend =</span> <span class="cn">FALSE</span></span>
<span id="cb27-25"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-25" tabindex="-1"></a>)</span>
<span id="cb27-26"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-26" tabindex="-1"></a></span>
<span id="cb27-27"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-27" tabindex="-1"></a>ht_list <span class="ot">&lt;-</span> ht1 <span class="sc">+</span> ht2</span>
<span id="cb27-28"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-28" tabindex="-1"></a></span>
<span id="cb27-29"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb27-29" tabindex="-1"></a><span class="fu">draw</span>(ht_list, <span class="at">heatmap_legend_side =</span> <span class="st">&quot;left&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-25-1.png" width="672" /></p>
<p>From this Heatmap we can see that generally the biological processes match the annotations, but we can also see that clusters 19 and 23 (on the left) show high expression of stress-related processes, while displaying low expression of the terms more related to radial glia, which is their annotation. This could explain why cluster 23 has such a low mapping confidence.</p>
<p>The same type of analysis can be done with the neurogenesis network:</p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-1" tabindex="-1"></a>sc_seu_neuro_GOBP <span class="ot">&lt;-</span> <span class="fu">do.call</span>(rbind, <span class="fu">lapply</span>(neurogenesis_GO, <span class="cf">function</span>(g) {</span>
<span id="cb28-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-2" tabindex="-1"></a>  </span>
<span id="cb28-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-3" tabindex="-1"></a>  g <span class="ot">&lt;-</span> g[<span class="fu">which</span>(g <span class="sc">%in%</span> <span class="fu">rownames</span>(sc_seu))]</span>
<span id="cb28-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-4" tabindex="-1"></a>  </span>
<span id="cb28-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-5" tabindex="-1"></a>  Matrix<span class="sc">::</span><span class="fu">colMeans</span>(<span class="fu">logcounts</span>(sc_pseudo)[g,])</span>
<span id="cb28-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-6" tabindex="-1"></a>  </span>
<span id="cb28-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-7" tabindex="-1"></a>}))</span>
<span id="cb28-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-8" tabindex="-1"></a></span>
<span id="cb28-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-9" tabindex="-1"></a><span class="fu">rownames</span>(sc_seu_neuro_GOBP) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">(GO:[0-9]+</span><span class="sc">\\</span><span class="st">)$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(sc_seu_neuro_GOBP))</span>
<span id="cb28-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb28-10" tabindex="-1"></a>sc_seu_neuro_GOBP <span class="ot">&lt;-</span> sc_seu_neuro_GOBP[,<span class="fu">mixedorder</span>(<span class="fu">colnames</span>(sc_seu_neuro_GOBP))]</span></code></pre></div>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-1" tabindex="-1"></a>neuro_gobp_pe <span class="ot">&lt;-</span> neurogenesis_preferential_GO<span class="sc">$</span>GO_Biological_Process_2025<span class="sc">$</span>activity[<span class="fu">names</span>(neurogenesis_GO),]</span>
<span id="cb29-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-2" tabindex="-1"></a></span>
<span id="cb29-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-3" tabindex="-1"></a>neuroTrends <span class="ot">&lt;-</span> <span class="fu">do.call</span>(cbind, <span class="fu">lapply</span>(<span class="fu">rownames</span>(neuro_gobp_pe), <span class="cf">function</span>(i) <span class="fu">smooth.spline</span>(<span class="dv">1</span><span class="sc">:</span><span class="fu">length</span>(neuro_gobp_pe[i,]), neuro_gobp_pe[i,], <span class="at">spar =</span> <span class="dv">1</span>)<span class="sc">$</span>y))</span>
<span id="cb29-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-4" tabindex="-1"></a><span class="fu">colnames</span>(neuroTrends) <span class="ot">&lt;-</span> <span class="fu">rownames</span>(neuro_gobp_pe)</span>
<span id="cb29-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-5" tabindex="-1"></a></span>
<span id="cb29-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-6" tabindex="-1"></a>neuroTrends <span class="ot">&lt;-</span> <span class="fu">t</span>(neuroTrends)[<span class="fu">names</span>(neurogenesis_GO),]</span>
<span id="cb29-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-7" tabindex="-1"></a>neuroTrends <span class="ot">&lt;-</span> neuroTrends[<span class="fu">order</span>(<span class="fu">apply</span>(neuroTrends, <span class="dv">1</span>, which.max), Matrix<span class="sc">::</span><span class="fu">rowMeans</span>(neuroTrends)),]</span>
<span id="cb29-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-8" tabindex="-1"></a></span>
<span id="cb29-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-9" tabindex="-1"></a><span class="fu">rownames</span>(neuroTrends) <span class="ot">&lt;-</span> <span class="fu">gsub</span>(<span class="st">&quot; </span><span class="sc">\\</span><span class="st">(GO:[0-9]+</span><span class="sc">\\</span><span class="st">)$&quot;</span>, <span class="st">&quot;&quot;</span>, <span class="fu">rownames</span>(neuroTrends))</span>
<span id="cb29-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-10" tabindex="-1"></a></span>
<span id="cb29-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-11" tabindex="-1"></a>stage_palette <span class="ot">&lt;-</span> <span class="fu">unique</span>(neurogenesis_sce<span class="sc">$</span>Stages_color)</span>
<span id="cb29-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-12" tabindex="-1"></a><span class="fu">names</span>(stage_palette) <span class="ot">&lt;-</span> <span class="fu">unique</span>(neurogenesis_sce<span class="sc">$</span>Stages)</span>
<span id="cb29-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-13" tabindex="-1"></a></span>
<span id="cb29-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-14" tabindex="-1"></a>h1 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(<span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(neuroTrends))), <span class="at">cluster_columns =</span> F, <span class="at">cluster_rows =</span> F, <span class="at">bottom_annotation =</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">df=</span><span class="fu">data.frame</span>(<span class="at">stage=</span>neurogenesis_sce<span class="sc">$</span>Stages), <span class="at">col=</span><span class="fu">list</span>(<span class="at">stage=</span>stage_palette), <span class="at">show_legend =</span> F, <span class="at">show_annotation_name =</span> F), <span class="at">name =</span> <span class="st">&#39;inferred expression</span><span class="sc">\n</span><span class="st">(z scaled)&#39;</span>, <span class="at">height =</span> <span class="fu">unit</span>(<span class="dv">100</span>, <span class="st">&#39;mm&#39;</span>), <span class="at">width =</span> <span class="fu">unit</span>(<span class="dv">30</span>, <span class="st">&#39;mm&#39;</span>))</span>
<span id="cb29-15"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-15" tabindex="-1"></a></span>
<span id="cb29-16"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-16" tabindex="-1"></a>ht2 <span class="ot">&lt;-</span> <span class="fu">Heatmap</span>(</span>
<span id="cb29-17"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-17" tabindex="-1"></a>  <span class="fu">t</span>(<span class="fu">scale</span>(<span class="fu">t</span>(sc_seu_neuro_GOBP[<span class="fu">rownames</span>(neuroTrends),]))),</span>
<span id="cb29-18"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-18" tabindex="-1"></a>  <span class="at">name =</span> <span class="st">&#39;scaled</span><span class="sc">\n</span><span class="st">expression&#39;</span>,</span>
<span id="cb29-19"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-19" tabindex="-1"></a>  <span class="at">width =</span> <span class="fu">unit</span>(<span class="fu">ncol</span>(sc_seu_neuro_GOBP) <span class="sc">*</span> <span class="fl">4.2</span>, <span class="st">&#39;mm&#39;</span>),</span>
<span id="cb29-20"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-20" tabindex="-1"></a>  <span class="at">cluster_columns =</span> <span class="cn">TRUE</span>, <span class="at">right_annotation =</span> <span class="fu">rowAnnotation</span>(<span class="at">max =</span> <span class="fu">anno_barplot</span>(<span class="fu">apply</span>(sc_seu_neuro_GOBP[<span class="fu">rownames</span>(neuroTrends),], <span class="dv">1</span>, max), <span class="at">gp =</span> <span class="fu">gpar</span>(<span class="at">fill =</span> <span class="st">&#39;black&#39;</span>))), <span class="at">bottom_annotation =</span> <span class="fu">HeatmapAnnotation</span>(<span class="at">annotation =</span> mapped_sc_pseudo<span class="sc">$</span>annotation<span class="sc">$</span>Best.Annotation[<span class="fu">colnames</span>(sc_seu_cortico_GOBP)], <span class="at">col =</span> <span class="fu">list</span>(<span class="at">annotation =</span> corticogenesis_palette), <span class="at">show_legend =</span> <span class="cn">FALSE</span>), <span class="at">show_column_dend =</span> <span class="cn">FALSE</span></span>
<span id="cb29-21"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-21" tabindex="-1"></a>)</span>
<span id="cb29-22"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-22" tabindex="-1"></a></span>
<span id="cb29-23"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-23" tabindex="-1"></a>ht_list <span class="ot">&lt;-</span> h1 <span class="sc">+</span> ht2</span>
<span id="cb29-24"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-24" tabindex="-1"></a></span>
<span id="cb29-25"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb29-25" tabindex="-1"></a><span class="fu">draw</span>(ht_list, <span class="at">heatmap_legend_side =</span> <span class="st">&quot;left&quot;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-27-1.png" width="672" /></p>
<p>Again, the cluster with the highest neuroinflammatory response (indicative of stress) is cluster 23. Here we can see that clusters that are assigned to immature neurons annotations indeed display higher expression of Biological Processes more associated with mature neurons compared to those clusters annotated as radial glia.</p>
<p>Additionally, we can visualize the mapped points directly on the eTraces, to see how similar they are to clusters that express more or less certain genes. The function to perform such analysis is <em>map_eTrace</em>, which requires the following inputs:</p>
<ul>
<li><code>net</code>: the reference network to use</li>
<li><code>mapped_obj</code>: the result of <em>mapNetwork</em></li>
<li>the inputs required to run <em>plot_eTrace</em> (see the chapter <code>3-Analysis_Tools</code>)</li>
</ul>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb30-1" tabindex="-1"></a><span class="fu">map_eTrace</span>(<span class="at">net =</span> corticogenesis_sce, <span class="at">mapped_obj =</span> mapped_sc_pseudo, <span class="at">genes =</span> <span class="st">&#39;VIM&#39;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-28-1.png" width="672" /></p>
<div class="sourceCode" id="cb31"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb31-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb31-1" tabindex="-1"></a><span class="fu">map_eTrace</span>(<span class="at">net =</span> corticogenesis_sce, <span class="at">mapped_obj =</span> mapped_sc_pseudo, <span class="at">genes =</span> <span class="st">&#39;SYT1&#39;</span>)</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-29-1.png" width="672" /></p>
<p>From this type on analysis we can clearly appreciate the distinction between radial glia-like clusters and more neuronal ones. Additionally, we can see at a glance which mapped points resemble more mature reference clusters, as age goes from left to right. The more rightward clusters are: 2, 9, 11, and 14.
If we check the age with the highest percentage in each cluster, we can see that those clusters are also those with the highest percentage of latest-stage cells:</p>
<div class="sourceCode" id="cb32"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb32-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-1" tabindex="-1"></a><span class="fu">sort</span>(<span class="fu">apply</span>(<span class="fu">table</span>(sc_seu<span class="sc">$</span>age, <span class="fu">paste0</span>(<span class="st">&#39;c&#39;</span>, sc_seu<span class="sc">$</span>seurat_clusters)), <span class="dv">2</span>, <span class="cf">function</span>(i) {i<span class="sc">/</span><span class="fu">sum</span>(i)})[<span class="st">&#39;Organoid-4M&#39;</span>,])</span>
<span id="cb32-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-2" tabindex="-1"></a><span class="co">#&gt;         c12         c13         c17         c20         c21 </span></span>
<span id="cb32-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-3" tabindex="-1"></a><span class="co">#&gt; 0.000000000 0.000000000 0.000000000 0.000000000 0.000000000 </span></span>
<span id="cb32-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-4" tabindex="-1"></a><span class="co">#&gt;         c22          c7         c16         c18         c10 </span></span>
<span id="cb32-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-5" tabindex="-1"></a><span class="co">#&gt; 0.000000000 0.000000000 0.002590674 0.002857143 0.008968610 </span></span>
<span id="cb32-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-6" tabindex="-1"></a><span class="co">#&gt;          c8          c3          c6          c5         c15 </span></span>
<span id="cb32-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-7" tabindex="-1"></a><span class="co">#&gt; 0.012219959 0.013736264 0.032374101 0.043478261 0.087281796 </span></span>
<span id="cb32-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-8" tabindex="-1"></a><span class="co">#&gt;         c19          c4          c1         c23          c2 </span></span>
<span id="cb32-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-9" tabindex="-1"></a><span class="co">#&gt; 0.097826087 0.103395062 0.111006585 0.117647059 0.428571429 </span></span>
<span id="cb32-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-10" tabindex="-1"></a><span class="co">#&gt;         c14         c11          c9 </span></span>
<span id="cb32-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb32-11" tabindex="-1"></a><span class="co">#&gt; 0.440097800 0.471655329 0.530303030</span></span></code></pre></div>
<p>We can also obtain markers for each cluster using the Seurat function <em>FindAllMarkers</em> and plot those genes in the eTraces.</p>
<div class="sourceCode" id="cb33"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb33-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-1" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">FindAllMarkers</span>(sc_seu, <span class="at">only.pos =</span> <span class="cn">TRUE</span>, <span class="at">min.pct =</span> <span class="fl">0.5</span>, <span class="at">logfc.threshold =</span> <span class="dv">1</span>)</span>
<span id="cb33-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-2" tabindex="-1"></a><span class="co">#&gt; Warning: `PackageCheck()` was deprecated in SeuratObject 5.0.0.</span></span>
<span id="cb33-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-3" tabindex="-1"></a><span class="co">#&gt; ℹ Please use `rlang::check_installed()` instead.</span></span>
<span id="cb33-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-4" tabindex="-1"></a><span class="co">#&gt; ℹ The deprecated feature was likely used in the Seurat</span></span>
<span id="cb33-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-5" tabindex="-1"></a><span class="co">#&gt;   package.</span></span>
<span id="cb33-6"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-6" tabindex="-1"></a><span class="co">#&gt;   Please report the issue at</span></span>
<span id="cb33-7"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-7" tabindex="-1"></a><span class="co">#&gt;   &lt;https://github.com/satijalab/seurat/issues&gt;.</span></span>
<span id="cb33-8"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-8" tabindex="-1"></a><span class="co">#&gt; This warning is displayed once every 8 hours.</span></span>
<span id="cb33-9"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-9" tabindex="-1"></a><span class="co">#&gt; Call `lifecycle::last_lifecycle_warnings()` to see where</span></span>
<span id="cb33-10"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-10" tabindex="-1"></a><span class="co">#&gt; this warning was generated.</span></span>
<span id="cb33-11"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-11" tabindex="-1"></a></span>
<span id="cb33-12"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-12" tabindex="-1"></a>markers <span class="ot">&lt;-</span> markers[<span class="fu">which</span>(markers<span class="sc">$</span>p_val_adj <span class="sc">&lt;</span> <span class="fl">0.001</span>),]</span>
<span id="cb33-13"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-13" tabindex="-1"></a></span>
<span id="cb33-14"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb33-14" tabindex="-1"></a>markers <span class="ot">&lt;-</span> <span class="fu">split</span>(markers<span class="sc">$</span>gene, markers<span class="sc">$</span>cluster)</span></code></pre></div>
<div class="sourceCode" id="cb34"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb34-1"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb34-1" tabindex="-1"></a><span class="fu">lapply</span>(<span class="fu">names</span>(markers), <span class="cf">function</span>(i) {</span>
<span id="cb34-2"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb34-2" tabindex="-1"></a>  g <span class="ot">&lt;-</span> markers[[i]]</span>
<span id="cb34-3"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb34-3" tabindex="-1"></a>  g <span class="ot">&lt;-</span> g[<span class="fu">which</span>(g <span class="sc">%in%</span> <span class="fu">rownames</span>(corticogenesis_sce))]</span>
<span id="cb34-4"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb34-4" tabindex="-1"></a>  <span class="fu">plot_eTrace</span>(<span class="at">net =</span> corticogenesis_sce, <span class="at">genes =</span> g, <span class="at">main =</span> i)</span>
<span id="cb34-5"><a href="mapping-a-single-cell-rnaseq-dataset.html#cb34-5" tabindex="-1"></a>})</span></code></pre></div>
<p><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-1.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-2.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-3.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-4.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-5.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-6.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-7.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-8.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-9.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-10.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-11.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-12.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-13.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-14.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-15.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-16.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-17.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-18.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-19.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-20.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-21.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-22.png" width="672" /><img src="4.1.1-Mapping_scRNAseq_files/figure-html/unnamed-chunk-32-23.png" width="672" /></p>
<pre><code>#&gt; [[1]]
#&gt; NULL
#&gt; 
#&gt; [[2]]
#&gt; NULL
#&gt; 
#&gt; [[3]]
#&gt; NULL
#&gt; 
#&gt; [[4]]
#&gt; NULL
#&gt; 
#&gt; [[5]]
#&gt; NULL
#&gt; 
#&gt; [[6]]
#&gt; NULL
#&gt; 
#&gt; [[7]]
#&gt; NULL
#&gt; 
#&gt; [[8]]
#&gt; NULL
#&gt; 
#&gt; [[9]]
#&gt; NULL
#&gt; 
#&gt; [[10]]
#&gt; NULL
#&gt; 
#&gt; [[11]]
#&gt; NULL
#&gt; 
#&gt; [[12]]
#&gt; NULL
#&gt; 
#&gt; [[13]]
#&gt; NULL
#&gt; 
#&gt; [[14]]
#&gt; NULL
#&gt; 
#&gt; [[15]]
#&gt; NULL
#&gt; 
#&gt; [[16]]
#&gt; NULL
#&gt; 
#&gt; [[17]]
#&gt; NULL
#&gt; 
#&gt; [[18]]
#&gt; NULL
#&gt; 
#&gt; [[19]]
#&gt; NULL
#&gt; 
#&gt; [[20]]
#&gt; NULL
#&gt; 
#&gt; [[21]]
#&gt; NULL
#&gt; 
#&gt; [[22]]
#&gt; NULL
#&gt; 
#&gt; [[23]]
#&gt; NULL</code></pre>
<p>Again, it is easy to distinguish between more neuronal clusters and more radial glia-like clusters, but also we can find some clusters that are more inhibitory like (e.g. clusters 22 and 11), and we can see that cluster 23 markers are also highly expressed in microglia, which again can point to possible stress.</p>

</div>
</div>
  </main>

  <div class="col-md-3 col-lg-2 d-none d-md-block sidebar sidebar-chapter">
    <nav id="toc" data-toggle="toc" aria-label="On this page">
      <h2>On this page</h2>
      <div id="book-on-this-page"></div>

      <div class="book-extra">
        <ul class="list-unstyled">
          <li><a id="book-source" href="#">View source <i class="fab fa-github"></i></a></li>
          <li><a id="book-edit" href="#">Edit this page <i class="fab fa-github"></i></a></li>
        </ul>
      </div>
    </nav>
  </div>

</div>
</div> <!-- .container -->

<footer class="bg-primary text-light mt-5">
  <div class="container"><div class="row">

  <div class="col-12 col-md-6 mt-3">
    <p>"<strong>4.1-Mapping_scRNAseq</strong>" was written by Erik Bot &amp; Asia Zonca. It was last built on 2025-11-12.</p>
  </div>

  <div class="col-12 col-md-6 mt-3">
    <p>This book was built by the <a class="text-light" href="https://bookdown.org">bookdown</a> R package.</p>
  </div>

</div></div>
</footer>


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    var src = "true";
    if (src === "" || src === "true") src = "https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/latest.js?config=TeX-MML-AM_CHTML";
    if (location.protocol !== "file:")
      if (/^https?:/.test(src))
        src = src.replace(/^https?:/, '');
    script.src = src;
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>
<script type="text/x-mathjax-config">const popovers = document.querySelectorAll('a.footnote-ref[data-toggle="popover"]');
for (let popover of popovers) {
  const div = document.createElement('div');
  div.setAttribute('style', 'position: absolute; top: 0, left:0; width:0, height:0, overflow: hidden; visibility: hidden;');
  div.innerHTML = popover.getAttribute('data-content');

  var has_math = div.querySelector("span.math");
  if (has_math) {
    document.body.appendChild(div);
    MathJax.Hub.Queue(["Typeset", MathJax.Hub, div]);
    MathJax.Hub.Queue(function() {
      popover.setAttribute('data-content', div.innerHTML);
      document.body.removeChild(div);
    })
  }
}
</script>
</body>

</html>
